src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,511 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+        const availableTarget = getNextAvailableTarget(existingTargets, diskParams.busType);
+        setDiskParams(diskParams => ({ ...diskParams, target: availableTarget }));
+    }, [vm.disks, diskParams.busType]);
 
-            if (value) {
-                this.setState({ verificationInProgress: true, validate: false });
-                cockpit.spawn(["head", "--bytes=16", value], { binary: true, err: "message", superuser: "try" })
-                        .then(file_header => {
-                            let format = "";
-
-                            // https://git.qemu.org/?p=qemu.git;a=blob;f=docs/interop/qcow2.txt
-                            if (file_header[0] == 81 && file_header[1] == 70 &&
-                                file_header[2] == 73 && file_header[3] == 251) {
-                                format = "qcow2";
-                                // All zeros, no backing file offset
-                                if (file_header.slice(8).every(bytes => bytes === 0)) {
-                                    this.setState({ customDiskVerificationFailed: false, format });
-                                } else {
-                                    this.setState({
-                                        format,
-                                        customDiskVerificationFailed: true,
-                                        customDiskVerificationMessage: _("Importing an image with a backing file is unsupported"),
-                                        validate: true
-                                    });
-                                }
+    useEffect(() => {
+        const file = diskParams.file;
+
+        if (file?.endsWith(".iso")) {
+            setDiskParams(diskParams => ({ ...diskParams, device: "cdrom" }));
+        }
+
+        if (file) {
+            setVerificationInProgress(true);
+            cockpit.spawn(["head", "--bytes=16", file], { binary: true, err: "message", superuser: "try" })
+                    .then(file_header => {
+                        let format = "";
+
+                        // https://git.qemu.org/?p=qemu.git;a=blob;f=docs/interop/qcow2.txt
+                        if (file_header[0] == 81 && file_header[1] == 70 &&
+                            file_header[2] == 73 && file_header[3] == 251) {
+                            format = "qcow2";
+                            // All zeros, no backing file offset
+                            if (file_header.slice(8).every(bytes => bytes === 0)) {
+                                setCustomDiskVerificationFailed(false);
                             } else {
-                                format = "raw";
-                                this.setState({ customDiskVerificationFailed: false, format });
+                                setCustomDiskVerificationFailed(true);
+                                setCustomDiskVerificationMessage(_("Importing an image with a backing file is unsupported"));
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231110-154059-d19bea72-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)